lib/python/bind_units.cpp

```diff
@@ -35,7 +96,28 @@ void init_units(py::module &m) {
       .def("__abs__", [](const units::Unit &self) { return abs(self); })
       .def(py::self == py::self)
       .def(py::self != py::self)
-      .def(hash(py::self));
+      .def(
+          "is_exactly_the_same",
+          [](const units::Unit &self, const units::Unit &other) {
+            return self.is_exactly_the_same(other);
+          },
+          "Check if two units are numerically identical.\n\n"
+          "The regular equality operator allows for small differences "
+          "in the unit's floating point multiplier. ``is_exactly_the_same`` "
+          "checks for exact identity.")
+      .def(hash(py::self))
+      .def("to_dict", to_dict,
+           "Serialize a unit to a dict.\n\nThis function is meant to be used "
+           "with :meth:`scipp.Unit.from_dict` to serialize units.\n\n"
+           "Warning\n"
+           "-------\n"
+           "The structure of the returned dict is an implementation detail and "
+           "may change without warning at any time! "
+           "It should not be used to access the internal representation of "
+           "``Unit``.")
```

Comment:
Are there performance implications for this when writing files? Have you considered returning a tuple of base-unit powers instead?